src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -4041,6 +4045,33 @@ private static bool IsNegativeConstantForArraySize(BoundExpression expression)
             {
                 analyzedArguments.Free();
             }
+
+            static void validateRecordCopyConstructor(NamedTypeSymbol containingType, MethodSymbol constructor, NamedTypeSymbol baseType, MethodSymbol baseCtor, Location errorLocation, DiagnosticBag diagnostics)
+            {
+                // Are we dealing with a user-defined copy constructor in a record type?
+                if (containingType is SourceNamedTypeSymbol sourceType &&
+                    sourceType.IsRecord &&
+                    SynthesizedRecordCopyCtor.HasCopyConstructorSignature(constructor))
+                {
+                    if (baseType.SpecialType == SpecialType.System_Object)
+                    {
+                        return;
+                    }
+
+                    // Unless the base type is 'object', the constructor should invoke a base type copy constructor
+                    HashSet<DiagnosticInfo> ignoredUseSiteDiagnostics = null;
+                    var foundBaseCopyCtor = SynthesizedRecordCopyCtor.FindCopyConstructor(baseType, containingType, ref ignoredUseSiteDiagnostics);
+                    if (foundBaseCopyCtor is null)
+                    {
+                        // if there is no base type copy constructor, we'll report that instead
+                        diagnostics.Add(ErrorCode.ERR_NoCopyConstructorInBaseType, errorLocation, baseType);
+                    }
+                    else if (baseCtor is null || !SynthesizedRecordCopyCtor.HasCopyConstructorSignature(baseCtor))
```

Comment:
>baseCtor is null [](start = 29, length = 16)

I do not think this method should ever be called with null ```baseCtor```

Reply:
See earlier comment and tests such as `CopyCtor_UserDefinedButDoesNotDelegateToBaseCopyCtor` that benefit from this diagnostic.

---
In reply to: [439015737](https://github.com/dotnet/roslyn/pull/45022#discussion_r439015737) [](ancestors = 439015737)